### PR TITLE
feat: cached data

### PR DESCRIPTION
**Pull request by DecathectZero**
attach a special dataAPI that caches versions/projects to the context for use through all the handlers

### CODE REVIEW

File: lib/controllers/interact.ts [hunk 1]
```diff
@@ -25,7 +26,7 @@ class InteractController extends AbstractController {
     const turn = new TurnBuilder<Context>(stateManager);
     turn.addHandlers(asr, nlu, dialog, runtime).addHandlers(tts, chips);
 
-    return turn.handle({ state, request, versionID, data: { locale } });
+    return turn.handle({ state, request, versionID, data: { locale } as any });
```
Comment (DecathectZero) [inline lib/controllers/interact.ts:29]: this is kinda a hack, need to update `@voiceflow/runtime` to make the init handler accept partial data

File: lib/services/state/index.ts [hunk 2]
```diff
@@ -52,13 +52,17 @@ class StateManager extends AbstractManager<{ utils: typeof utils }> implements I
     if (!context.versionID) {
       throw new Error('context versionID not defined');
     }
+
+    // cache per interaction (save version call during request/response cycle)
+    const api = new CacheDataAPI(this.services.dataAPI);
```
Comment (DecathectZero) [inline lib/services/state/index.ts:57]: this is the most important thing that ties everything together

File: lib/controllers/interact.ts (hunk 1 above)
Comment (DecathectZero) [reply lib/controllers/interact.ts]: done this now

**Review by leartgjoni-voiceflow — APPROVED** [review]
solid

**Pull request merged**